ompi/mca/coll/basic/coll_basic_neighbor_allgatherv.c

```diff
@@ -64,36 +63,40 @@ mca_coll_basic_neighbor_allgatherv_cart(const void *sbuf, int scount, struct omp
         }
 
         if (MPI_PROC_NULL != srank) {
+            nreqs += 2;
```

Comment:
@bosilca Replace nreqs+=2 by nreqs++ before each request. That way, we'll be able to
free axactly the right number of requests in case of failure.